### PR TITLE
Fix breadcrumbs

### PR DESCRIPTION
**Pull request by tjol**
Re #311 — truncate the path if there isn't enough room to show it.

### CODE REVIEW

File: sunflower/gui/main_window.py [hunk 1]
```diff
@@ -2480,6 +2480,7 @@ def register_class(self, name, title, PluginClass):
 
 		# create action
 		action = Gio.SimpleAction.new(name, None)
+		action.plugin_class = PluginClass
```
Comment (MeanEYE) [inline sunflower/gui/main_window.py]: Can you be more descriptive with changes? Why was this done and how it resulted in a fix?

**Comment by MeanEYE** [comment]
Your commit messages are not very helpful. It's a mish-mash of questions, Vim syntax, description, etc. More to the point there's no description in commits so I have no idea why you made changes and what they fixed. 

Can you please provide more information so I know how this should be treated? Also, does this fix depend on set of patches that port Sunflower to Python 3, if so, then we will branch out on v2 vs v3 which is not something I wish to happen.

**Comment by tjol** [comment]
This fix (limited to the last commit here) shouldn't depend on the Python 3 changes, but that's what I was working from.

I could do some history rewriting if you like, if you give me some guidance on ‘commit philosophy’. If restrospectively every commit should result in a working copy, it might make most sense to squash *all* commits in the python3 branch into one. Or I could split them out by theme – with a commit changing all calls to `map` and `filter` that need changing. Not sure what would be best.

File: sunflower/gui/main_window.py (hunk 1 above)
Comment (tjol) [reply sunflower/gui/main_window.py]: This fixed a crash at (IIRC) https://github.com/MeanEYE/Sunflower/blob/1aeeaff6439759867adb6d09a0094daa99429947/sunflower/gui/main_window.py#L924) – I don't know if that was Python 3 specific

**Comment by tjol** [comment]
As to how this should be treated: I'd suggest looking at the Python 3 PR first. I'm happy to answer any questions about specifics, and we can make any changes that still need to be made there, be it for compatibility, due to additional bugs found, or for more readable history. *Then* I'll rebase this. and it'll all make more sense.

**Comment by MeanEYE** [comment]
Ideally we should have a branch per feature, all based on development branch. That way we can easily merge one and not worry about the other.

As for PR for Pythong3, I already replied to that. Thank you for these PRs, they are much appreciated. I say this as I don't want anyone getting the feeling I am slacking on merges. Just that work is taking a huge chunk of my life. However, these might be a wake up call I needed for a while when it comes to personal projects.

As for this PR. I will pull it in separate branch and probably just cherry-pick the last commit or few.

File: sunflower/gui/main_window.py (hunk 1 above)
Comment (MeanEYE) [reply sunflower/gui/main_window.py]: Okay, thanks. Makes sense.

**Comment by MeanEYE** [comment]
Am closing this pull-request. We cherry-picked some of the fixes and breadcrumbs will receive overhaul since new Nautilus has different look and layout to the previous one and we kind of have to follow that to a degree to keep the looks consistent.

**Pull request closed without merging**